v2/internal/frontend/desktop/windows/frontend.go

```diff
@@ -689,6 +696,73 @@ func (f *Frontend) processMessage(message string) {
 	}()
 }
 
+func (f *Frontend) processMessageWithAdditionalObjects(message string, sender *edge.ICoreWebView2, args *edge.ICoreWebView2WebMessageReceivedEventArgs) {
+	if strings.HasPrefix(message, "file:") {
+		callbackID := message[5:]
+
+		objs, err := args.GetAdditionalObjects()
+		defer objs.Release()
+
+		if err != nil {
+			f.logger.Error(err.Error())
+			return
+		}
+
+		if objs == nil {
+			return
+		}
+
+		count, err := objs.GetCount()
+		if err != nil {
+			f.logger.Error(err.Error())
+			return
+		}
+
+		files := []File{}
+		for i := uint32(0); i < count; i++ {
+			_file, err := objs.GetValueAtIndex(i)
+			defer func() {
+				err := _file.CallRelease(unsafe.Pointer(_file))
+				if err != nil {
+					f.logger.Error("cannot release resource: %s", err.Error())
+				}
+			}()
+			if err != nil {
+				f.logger.Error("cannot get value at %d : %s", i, err.Error())
+				return
+			}
+
+			file := (*edge.ICoreWebView2File)(unsafe.Pointer(_file))
+			defer file.Release()
```

Comment:
Ah we have this release here. That may end up being a double release.

Reply:
I have a question, is this casting will create a new allocation or it will reuse the previous allocation (returned by `GetValueAtIndex`)? 

If it is reused, then can we remove the defer function above, as it will be released here?

Reply:
This is just a pointer conversion as such, the reference counter will not be increased and you don't need to call `Release` an additional time. In contrast if you use `IUnknown.QueryInterface` you would need to call `Release` on both `_file` and `file`.